### PR TITLE
Fix Import_shipping_carriers method

### PR DESCRIPTION
**Pull request by cerebro1**

### CODE REVIEW

File: channel.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def get_current_magento_channel(cls):
         return channel
 
     @classmethod
-    def import_shipping_carriers(cls, channels):
+    def import_shipping_carriers(self, channels):
```
Comment (riteshshrv) [inline channel.py]: revert

**Comment by cerebro1** [comment]
https://3.basecamp.com/3194017/buckets/421793/todos/113505435

File: channel.py [hunk 2]
```diff
@@ -231,7 +231,8 @@ def import_shipping_carriers(cls, channels):
         SaleChannelCarrier = Pool().get('sale.channel.carrier')
 
         for channel in channels:
-            assert channel.source == 'magento'
+            if self.source == 'magento':
```
Comment (riteshshrv) [inline channel.py]: `channel.source`

File: channel.py [hunk 3]
```diff
@@ -231,7 +231,8 @@ def import_shipping_carriers(cls, channels):
...
+                super(Channel, self).import_shipping_carriers()
```
Comment (riteshshrv) [inline channel.py]: `carriers = super(Channel, self).import_shipping_carriers()`

File: channel.py [hunk 4]
```diff
@@ -231,7 +231,8 @@ def import_shipping_carriers(cls, channels):
         SaleChannelCarrier = Pool().get('sale.channel.carrier')
 
         for channel in channels:
-            assert channel.source == 'magento'
+            if channel.source == 'magento':
```
Comment (tarunbhardwaj) [inline channel.py]: Wrong.
Call super for all non magento listings  together before for loop.

File: channel.py [hunk 5]
```diff
@@ -240,7 +241,7 @@ def import_shipping_carriers(cls, channels):
                 ) as order_config_api:
                     carriers_data = order_config_api.get_shipping_methods()
 
-            carriers = []
+            carriers = carriers or []
```
Comment (tarunbhardwaj) [inline channel.py]: not needed

File: channel.py [hunk 6]
```diff
@@ -231,7 +231,8 @@ def import_shipping_carriers(cls, channels):
         SaleChannelCarrier = Pool().get('sale.channel.carrier')
 
         for channel in channels:
```
Comment (tarunbhardwaj) [inline channel.py]: and make sure this loop is just called for magento listings

File: channel.py [hunk 7]
```diff
@@ -222,17 +222,18 @@ def get_current_magento_channel(cls):
         return channel
 
     @classmethod
```
Comment (riteshshrv) [inline channel.py]: remove this

File: channel.py [hunk 8]
```diff
@@ -222,17 +222,18 @@ def get_current_magento_channel(cls):
...
-    def import_shipping_carriers(cls, channels):
+    def import_shipping_carriers(self, channels):
```
Comment (riteshshrv) [inline channel.py]: only one argument `self`

File: channel.py [hunk 9]
```diff
@@ -222,17 +222,18 @@ def get_current_magento_channel(cls):
...
         :param channels: Active record list of magento channels
         """
         SaleChannelCarrier = Pool().get('sale.channel.carrier')
 
-        for channel in channels:
-            assert channel.source == 'magento'
+        if self.source != 'magento':
+            carriers = super(Channel, self).import_shipping_carriers()
 
+        for channel in channels:
```
Comment (riteshshrv) [inline channel.py]: no `for` loop

File: channel.py [hunk 10]
```diff
@@ -221,43 +221,42 @@ def get_current_magento_channel(cls):
...
-            carriers = []
-            for data in carriers_data:
-                carrier = SaleChannelCarrier.search([
-                    ('code', '=', data['code']),
-                    ('channel', '=', channel.id),
-                ])
-                if carrier:
-                    continue
+        if self.source != 'magento':
+            carriers = super(Channel, self).import_shipping_carriers()
```
Comment (riteshshrv) [inline channel.py]: `carriers = ` is not required

Comment (tarunbhardwaj) [inline channel.py]: return

**Comment by cerebro1** [comment]
for magento:
magento_error

for nonmagento
error_channel

**Comment by riteshshrv** [comment]
LGTM

**Pull request closed without merging**